pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1177,18 +1177,30 @@ func (r *ClickHouseReader) GetSpanFilters(ctx context.Context, queryParams *mode
 				traceFilterReponse.Status = map[string]uint64{"ok": 0, "error": 0}
 			}
 		case constants.Duration:
-			finalQuery := fmt.Sprintf("SELECT durationNano as numTotal FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.durationTable)
+			err := r.featureFlags.CheckFeature(constants.DurationSort)
+			durationSortEnabled := err == nil
+			finalQuery := ""
+			if !durationSortEnabled {
+				finalQuery = fmt.Sprintf("SELECT durationNano as numTotal FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.indexTable)
+			} else {
+				finalQuery = fmt.Sprintf("SELECT durationNano as numTotal FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.durationTable)
+			}
 			finalQuery += query
 			finalQuery += " ORDER BY durationNano LIMIT 1"
 			var dBResponse []model.DBResponseTotal
-			err := r.db.Select(ctx, &dBResponse, finalQuery, args...)
+			err = r.db.Select(ctx, &dBResponse, finalQuery, args...)
 			zap.S().Info(finalQuery)
 
 			if err != nil {
 				zap.S().Debug("Error in processing sql query: ", err)
 				return nil, &model.ApiError{Typ: model.ErrorExec, Err: fmt.Errorf("Error in processing sql query: %s", err)}
 			}
-			finalQuery = fmt.Sprintf("SELECT durationNano as numTotal FROM %s.%s WHERE timestamp >= @timestampL AND timestamp <= @timestampU", r.TraceDB, r.durationTable)
+
+			if !durationSortEnabled {
```

Comment:
@makeavish why are we repeating this again here after L1183?

Reply:
We make two DB queries to fetch min and max duration

Reply:
We make two DB queries to fetch min and max duration

Reply:
using `min(durationNano)` and `max(durationNano)` in a single query not feasible?

Reply:
No, it's not efficient.
As we are sorting by durationNano and limiting result to 1. This is more efficient than using `min()` clickhouse function

Reply:
But in case where we are using main table it might be more efficient to fetch min and max in single call if skip index is not helpful in case of order by

Reply:
yes..I referring for the case when `durationSortEnabled` is `false`.

Reply:
Done